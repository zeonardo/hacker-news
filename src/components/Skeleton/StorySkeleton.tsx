import React from 'react';
import { Box, Skeleton, Stack } from '@mui/material';

interface StorySkeletonProps {
  itemCount?: number;
}

const skeletonTheme = { bgcolor: 'grey.500' };

const StorySkeleton: React.FC<StorySkeletonProps> = ({ itemCount = 1 }) => (
  <Stack
    direction="row"
    alignItems="stretch"
    minHeight={100}
    spacing={2}
    sx={{ overflowX: 'auto', paddingX: 2, bgcolor: 'black' }}
  >
    {Array.from({ length: itemCount }).map((_, index) => (
      <Box
        key={`skeleton-${index}`}
        minWidth={300}
        width="100%"
        py={'.5rem'}
        px={2}
        sx={{ maxWidth: '100%', paddingY: 20, bgcolor: '#000' }}
      >
        <Skeleton sx={{ ...skeletonTheme }} variant="text" width="100%" height={'2rem'} />
        <Skeleton sx={{ ...skeletonTheme }} variant="text" width="80%" height={'2rem'} />
        <Box display="flex" justifyContent={'space-between'} my={1}>
          <Skeleton sx={{ ...skeletonTheme }} variant="text" width="50%" />
          <Skeleton sx={{ ...skeletonTheme }} variant="text" width="30%" />
        </Box>
        <Box display="flex" justifyContent={'space-between'}>
          <Skeleton sx={{ ...skeletonTheme }} variant="text" width="40%" />
          <Skeleton sx={{ ...skeletonTheme }} variant="text" width="30%" />
        </Box>
      </Box>
    ))}
  </Stack>
);

export default StorySkeleton;
