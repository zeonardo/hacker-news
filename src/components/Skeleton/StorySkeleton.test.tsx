import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import StorySkeleton from './StorySkeleton';

describe('StorySkeleton component', () => {
  it('renders without crashing', () => {
    render(<StorySkeleton />);
    expect(document.body).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<StorySkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
