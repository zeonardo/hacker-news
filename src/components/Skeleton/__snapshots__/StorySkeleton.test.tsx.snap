// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StorySkeleton component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1vkoruo-MuiStack-root"
  >
    <div
      class="MuiBox-root css-1mc38uo"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ejrkhc-MuiSkeleton-root"
        style="width: 100%; height: 2rem;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ejrkhc-MuiSkeleton-root"
        style="width: 80%; height: 2rem;"
      />
      <div
        class="MuiBox-root css-1pup0r6"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ejrkhc-MuiSkeleton-root"
          style="width: 50%;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ejrkhc-MuiSkeleton-root"
          style="width: 30%;"
        />
      </div>
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ejrkhc-MuiSkeleton-root"
          style="width: 40%;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ejrkhc-MuiSkeleton-root"
          style="width: 30%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
